src/test_report.py

```diff
@@ -108,39 +92,18 @@ def _get_results_data(self, root_node):
             'groups': groups,
         }
 
-    def _create_test_report(self, root_node):
-        results = self._get_results_data(root_node)
-        template_env = jinja2.Environment(
-                            loader=jinja2.FileSystemLoader("./config/reports/")
-                        )
-        template = template_env.get_template("test-report.jinja2")
-        revision = root_node['revision']
-        stats = results['stats']
-        groups = results['groups']
-        subject = (f"\
-{revision['tree']}/{revision['branch']} {revision['describe']}: \
-{stats['total']} runs {stats['failures']} failures")
-        content = template.render(
-            subject=subject, root=root_node, groups=groups
-        )
-        return content, subject
-
-    def _send_email(self, content, subject):
-        smtp = self._smtp_connect()
-        if smtp:
-            email_msg = self._create_email(
-                self._email_send_to, self._email_send_from,
-                subject, content
-            )
-            smtp.send_message(email_msg)
-            smtp.quit()
-
-    def _make_report(self, root_node, dump=True, send=False):
+    def _get_report(self, root_node, dump=True, send=False):
```

Comment:
I find the method name confusing  `_get_report` suggests that it actually will return something while it doesn't.
It actually does several things: it creates report content (and subject), it sends it over e-mail or dumps it to stdout. I'd go with refactoring the code so we can have separate logic that creates the content and separate which sends it.

Reply:
Thanks for pointing that out.
I have created two separate methods to dump and send reports.
Creation of email will directly take place from `email_sender` now.